### PR TITLE
Improve testing of concurrent query manager

### PR DESCRIPTION
**Pull request by khanaffan**
* Splitting test into more atomic test.
* Fix a potential hang issue.

ITwinJs PR: https://github.com/iTwin/itwinjs-core/pull/4667

TS: 4682

### CODE REVIEW

**Comment by pmconne** [comment]
@khanaffan is this at all related to https://github.com/iTwin/itwinjs-core/issues/4355? That issue is about a TypeScript test.

**Comment by khanaffan** [comment]
> @khanaffan is this at all related to [iTwin/itwinjs-core#4355](https://github.com/iTwin/itwinjs-core/issues/4355)? That issue is about a TypeScript test.

Yes, it introduces a test parameter called `delay` that prevent query to be executed before that time. Which ensure the test query can be cancelled every time.

**Comment by pmconne** [comment]
> Yes, it introduces a test parameter called delay that prevent query to be executed before that time

Does the TypeScript test in itwinjs-core need to be updated to make use of that parameter? If so, is PR filed?

**Comment by khanaffan** [comment]
> > Yes, it introduces a test parameter called delay that prevent query to be executed before that time
> 
> Does the TypeScript test in itwinjs-core need to be updated to make use of that parameter? If so, is PR filed?

No i have not created PR yet. I wanted to add documentation around concurrent query classes as requested by someone on git hub issues. I will publish it by end of today.

**Comment by khanaffan** [comment]
@pmconne  Here is the other PR https://github.com/iTwin/itwinjs-core/pull/4667

File: iModelCore/ECDb/ECDb/ConcurrentQueryManagerImpl.cpp [hunk 1]
```diff
@@ -16,7 +17,40 @@ using namespace std::chrono;
...
+#else
+    #define log_trace(...) 
+    #define log_debug(...) 
+#endif
+
+#define log_info(...)  s_logger.infov(__VA_ARGS__)
+#define log_warn(...)  s_logger.warningv(__VA_ARGS__)
+#define log_error(...) s_logger.errorv(__VA_ARGS__)
+
+static const char* GetTimestamp() {
```
[the remark this answers is not in the document]
Comment (khanaffan) [reply iModelCore/ECDb/ECDb/ConcurrentQueryManagerImpl.cpp]: I switch to using `DateTime::GetCurrentTime().ToTimestampString`.

**Review by pmconne — APPROVED** [review]

**Review by ColinKerr — APPROVED** [review]

File: iModelCore/ECDb/ECDb/ConcurrentQueryManagerImpl.h [hunk 2]
```diff
@@ -11,14 +11,16 @@
 #include <random>
 #include <chrono>
 
+#define DEFAULT_QUERY_DELAY_MAX_TIME    std::chrono::seconds(10) // 20 sec
```
Comment (kabentley) [inline iModelCore/ECDb/ECDb/ConcurrentQueryManagerImpl.h:14]: Either the comment or the code is wrong

Comment (khanaffan) [reply iModelCore/ECDb/ECDb/ConcurrentQueryManagerImpl.h]: I removed the comments.

**Pull request merged**